app/controllers/shops_controller.rb

```diff
@@ -0,0 +1,50 @@
+class ShopsController < ApplicationController
+  def show
+    @shop_owner = ShopOwner.find(params[:shop_owner_id])
+    @shop = @shop_owner.shop
+  end
+
+  def new
+    @shop_owner = ShopOwner.find(params[:shop_owner_id])
+    @shop = Shop.new
+  end
+
+  def create
+    @shop_owner = ShopOwner.find(params[:shop][:shop_owner_id])
+    @shop = Shop.new(shop_params)
+    if @shop.save
+      @shop_owner.update(shop: @shop)
+      redirect_to dashboard_path(@shop_owner), notice: "Shop Creation Successful"
+      session[:shop_owner_id] = @shop_owner.id
+    else
+      flash["errors"] = @shop.errors.full_messages
+      render :new
+    end
+  end
+
+  def update
+    respond_to do |format|
+      if @shop.update(shop_params)
+        format.html { redirect_to @shop, notice: "Shop was successfully updated." }
+        format.json { render :show, status: :ok, location: @shop }
+      else
+        format.html { render :edit }
+        format.json { render json: @shop.errors, status: :unprocessable_entity }
```

Comment:
This is cool. Are we building now with API in mind? Or is there some added benefit for rendering JSON?